app/src/main/java/pl/szczodrzynski/edziennik/ui/modules/settings/contributors/ContributorsActivity.kt

```diff
@@ -35,6 +36,61 @@ class ContributorsActivity : AppCompatActivity(), CoroutineScope {
     // local/private variables go here
     private val errorSnackbar: ErrorSnackbar by lazy { ErrorSnackbar(this) }
 
+    var konami = 0
+
+    override fun onKeyUp(keyCode: Int, event: KeyEvent): Boolean {
+        return when (keyCode) {
+            KeyEvent.KEYCODE_DPAD_UP -> {
+                if (konami == 0 || konami == 1)
+                    konami += 1
+                else
+                    konami = 0
+                true
+            }
+            KeyEvent.KEYCODE_DPAD_DOWN -> {
+                if (konami == 2 || konami == 3)
+                    konami += 1
+                else
+                    konami = 0
+                true
+            }
+            KeyEvent.KEYCODE_DPAD_LEFT-> {
+                if (konami == 4 || konami == 6)
+                    konami += 1
+                else
+                    konami = 0
+                true
+            }
+            KeyEvent.KEYCODE_DPAD_RIGHT -> {
+                if (konami == 5 || konami == 7)
+                    konami += 1
+                else
+                    konami = 0
+                true
+            }
+            KeyEvent.KEYCODE_B -> {
+                if (konami == 8)
+                    konami += 1
+                else
+                    konami = 0
+                true
+            }
+            KeyEvent.KEYCODE_A -> {
+                if (konami == 9) {
+                    konami += 1
+                    b.konami.isVisible = true
+                }
+                else
+                    konami = 0
+                true
+            }
+            else -> {
+                konami = 0
+                super.onKeyUp(keyCode, event)
+            }
+        }
+    }
```

Comment:
This is extreeemally long.. isn't there any easier, or shorter way to write this algorithm?

Reply:
There is a shorter way

Reply:
Cool, I'm waiting then.